src/System.Net.Security/src/System/Net/_SecureChannel.cs

```diff
@@ -0,0 +1,1487 @@
+                                eppStruct.pwszServerName = namePtr;
+                                cppStruct.dwFlags |=
+                                    (Interop.Crypt32.CertChainPolicyIgnoreFlags.CERT_CHAIN_POLICY_IGNORE_ALL &
+                                     ~Interop.Crypt32.CertChainPolicyIgnoreFlags.CERT_CHAIN_POLICY_IGNORE_INVALID_NAME_FLAG);
+
+                                SafeX509ChainHandle chainContext = chain.SafeHandle;
+                                status = Verify(chainContext, ref cppStruct);
+                                if (status == Interop.Crypt32.CertChainPolicyErrors.CERT_E_CN_NO_MATCH)
+                                    sslPolicyErrors |= SslPolicyErrors.RemoteCertificateNameMismatch;
+                            }
+                        }
+                    }
+
+                    X509ChainStatus[] chainStatusArray = chain.ChainStatus;
+                    if (chainStatusArray != null && chainStatusArray.Length != 0)
+                        sslPolicyErrors |= SslPolicyErrors.RemoteCertificateChainErrors;
+                }
+
+                if (remoteCertValidationCallback != null)
+                {
+                    success = remoteCertValidationCallback(_hostName, remoteCertificateEx, chain, sslPolicyErrors);
+                }
+                else
+                {
+                    if (sslPolicyErrors == SslPolicyErrors.RemoteCertificateNotAvailable && !_remoteCertRequired)
+                        success = true;
+                    else
+                        success = (sslPolicyErrors == SslPolicyErrors.None);
+                }
+
+                if (Logging.On)
+                {
+                    if (sslPolicyErrors != SslPolicyErrors.None)
+                    {
+                        Logging.PrintInfo(Logging.Web, this, SR.net_log_remote_cert_has_errors);
+                        if ((sslPolicyErrors & SslPolicyErrors.RemoteCertificateNotAvailable) != 0)
+                            Logging.PrintInfo(Logging.Web, this, "\t" + SR.net_log_remote_cert_not_available);
+                        if ((sslPolicyErrors & SslPolicyErrors.RemoteCertificateNameMismatch) != 0)
+                            Logging.PrintInfo(Logging.Web, this, "\t" + SR.net_log_remote_cert_name_mismatch);
+                        if ((sslPolicyErrors & SslPolicyErrors.RemoteCertificateChainErrors) != 0)
+                            foreach (X509ChainStatus chainStatus in chain.ChainStatus)
+                                Logging.PrintInfo(Logging.Web, this, "\t" + chainStatus.StatusInformation);
+                    }
+                    if (success)
+                    {
+                        if (remoteCertValidationCallback != null)
+                            Logging.PrintInfo(Logging.Web, this, SR.net_log_remote_cert_user_declared_valid);
+                        else
+                            Logging.PrintInfo(Logging.Web, this, SR.net_log_remote_cert_has_no_errors);
```

Comment:
Well, it could have had "an error" of RemoteCertificateNotAvailable which can sometimes be considered OK, and would have just printed as "cert has errors".

Didn't bother to look up what would appear in the logs, but the name of the string implies something healthier than the minimums here.